terraform/sgtm-cluster/main.tf

```diff
@@ -407,6 +408,43 @@ resource "aws_lambda_permission" "lambda_permission_for_sgtm_rest_api" {
   source_arn = "${var.sgtm_rest_api_execution_arn}/*/${aws_api_gateway_method.sgtm_post.http_method}${aws_api_gateway_resource.sgtm_resource.path}"
 }
 
+## SQS
+resource "aws_sqs_queue" "sgtm-webhooks-queue-fifo" {
+  name = "sgtm-webhooks-queue${local.suffix}.fifo"
+  fifo_queue = true
+  content_based_deduplication = true
```

Comment:
This is AWS's content dedupe logic right? Should protect against duplicate or similar enough webhooks but were we thinking of doing more advanced deduping? Is there a way to specify a dedupe key/logic?

Reply:
Yep, messages are deduped by the `MessageGroupId` which is currently being set to the webhook `delivery_id`. We were thinking of deduping by PR node id, but would need custom logic to avoid losing different webhook event types. I believe aws does not allow custom deduping logic.

Reply:
SGTM is in a weird spot right now where much of the event handler logic is the same except for a few extra steps or queries or datamodel changes. All events that use the same pull request node id end up waiting for the same lock because these handlers have slightly divergent but shared mutations. 

I think it would be interesting to see if we could isolate the event handler's logic more and be able to lock on different node ids (i.e. updating a PR comment does not lock the PR node id -- although not sure if asana would take that well). Something to look into when we have the bandwidth 😅